bundles/org.eclipse.text.quicksearch/src/org/eclipse/text/quicksearch/internal/core/QuickTextSearcher.java

```diff
@@ -121,7 +123,27 @@ private SearchInFilesWalker createWalker(PriorityFunction priorities) {
 		job.schedule();
 		return job;
 	}
-
+	/**
+	 * Try to get a content as String. Avoids Streaming.
+	 */
+	private static String toShortString(IFile file) {
+		/**
+		 * Just any number such that the most source files will fit in. And not too
+		 * big to avoid out of memory.
+		 **/
+		int MAX_BUFFER_LENGTH = 999_999; // max 2MB.
+		try {
+			byte[] content = file.readNBytes(MAX_BUFFER_LENGTH);
+			int length = content.length;
+			if (length >= MAX_BUFFER_LENGTH) {
```

Comment:
This looks strange, why should the method read more than requested? In any case the data is there why not use it? And why then not better check for length <= 0 ?

the javadoc says:

```
	 * @param maxBytes The maximal length of the returned array. If maxBytes is
	 *                 greater or equal to the file length the whole content is
	 *                 returned. If maxBytes is smaller then the file length then
	 *                 only the first maxBytes bytes are returned.
```

so this will never return a larger array. looking furhter down, it feels like you first want to check the file-size.

Reply:
By the way even though such method is missing in the API if used in a loop like here it is often better to allocate an array once and reuse that array, like in `java.io.InputStream.readNBytes(byte[], int, int)`